procrastinate/job_context.py

```diff
@@ -34,7 +34,7 @@ def as_dict(self):
         return result
 
 
-@attr.dataclass(frozen=True, kw_only=True)
+@attr.dataclass(frozen=True, kw_only=True, eq=False)
```

Comment:
The job context is used as a dictionary key to keep track of pending job tasks (where the task is the value)

Reply:
yes, but I would have thought being able to be used as a dict key included having a `__eq__` method, otherwise how do you distinguish different objects with the same hash ?

Reply:
Removed `eq=False` based on other suggestion